nomad/job_endpoint_hook_vault.go

```diff
@@ -34,7 +34,7 @@ func (h jobVaultHook) Validate(job *structs.Job) ([]error, error) {
 	for _, tg := range vaultBlocks {
 		for _, vaultBlock := range tg {
 			vconf := h.srv.config.VaultConfigs[vaultBlock.Cluster]
-			if !vconf.IsEnabled() {
+			if vconf == nil || !vconf.IsEnabled() {
```

Comment:
I'd probably move this into the `IsEnabled` method, where we're nil-checking the `*bool` already but missed nil-checking the struct itself. That'll prevent this problem from happening anywhere else we check that method in the future.